### DISCUSSION TITLE
Is it possible to use Radix's collision detection capabilities with Framer Motion?

### DISCUSSION THREAD

**Discussion by LeonLiu2020** [radix-ui/primitives#2958] [Help]
I am using Framer Motion and Radix Primitives, and I found that most tutorials and examples online use the forceMount method and manage states manually. However, this approach seems to lose the collision detection capability of Radix Primitives. For example, it's relatively easy to respond to data attribute changes through CSS animations:
`.TooltipContent[data-side='top'] {
  animation-name: slideUp;
}
.TooltipContent[data-side='bottom'] {
  animation-name: slideDown;
}`

But if I want to achieve the same logic using Framer Motion, I haven't found a straightforward way. It seems that Framer Motion cannot easily respond to changes in data attributes?

In other words, the examples using forceMount and manually managing states do not have collision detection capabilities?
